flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/CheckpointExceptionHandlerConfigurationTest.java

```diff
@@ -75,6 +79,19 @@ public CheckpointExceptionHandler createCheckpointExceptionHandler(
 			}
 		};
 
+		if (failOnException) {
+			CheckpointExceptionHandler exceptionHandler =
+				inspectingFactory.createCheckpointExceptionHandler(failOnException, environment);
+			Assert.assertTrue(
+				exceptionHandler instanceof CheckpointExceptionHandlerFactory.FailingCheckpointExceptionHandler
+			);
+
+			CheckpointExceptionHandlerFactory.FailingCheckpointExceptionHandler actuallyHandler =
+				(CheckpointExceptionHandlerFactory.FailingCheckpointExceptionHandler) exceptionHandler;
+
+			Assert.assertEquals(3, actuallyHandler.tolerableNumber);
```

Comment:
I would rather make fields of `FailingCheckpointExceptionHandler` private and check `tolerableNumber` the same way as `failTaskOnCheckpointException` in previous declaration of `CheckpointExceptionHandlerFactory`